packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -277,26 +277,29 @@ const makeRenderer = (
 		};
 		const checkReady = state =>
 			state.preRenderChecklist.every(isReady => isReady);
-		const populateStore = store =>
-			new Promise((resolve, reject) => {
-				// dispatch SERVER_RENDER to kick off API middleware
-				store.dispatch({ type: SERVER_RENDER, payload: url });
+		const populateStore = store => {
+			// keep this above the SERVER_RENDER so
+			// state.flags is available to query calls
+			if (checkReady(store.getState())) {
+				return addFlags(store).then(() => {
+					// dispatch SERVER_RENDER to kick off API middleware
+					store.dispatch({ type: SERVER_RENDER, payload: url });
 
-				if (checkReady(store.getState())) {
-					addFlags(store).then(() => {
-						resolve(store);
-					});
-					return;
-				}
-				const unsubscribe = store.subscribe(() => {
-					if (checkReady(store.getState())) {
-						addFlags(store).then(() => {
-							resolve(store);
-							unsubscribe();
+					new Promise((resolve, reject) => {
```

Comment:
this will have to be `return`ed in order to prevent the outer Promise from resolving